src/types/form.ts

```diff
@@ -180,6 +180,18 @@ export type UseFormGetValues<TFieldValues extends FieldValues> = {
   ): [...FieldPathValues<TFieldValues, TFieldNames>];
 };
 
+export type _UseFormGetFieldState<TFieldValues extends FieldValues> = <
+  TFieldName extends FieldPath<TFieldValues>,
+>(
+  name: TFieldName,
+  formState?: FormState<TFieldValues>,
+) => {
+  invalid: boolean;
+  isDirty: boolean;
+  isTouched: boolean;
+  error: FieldError;
+};
```

Comment:
I think there is an error in the type of the return value.
The type of a property other than invalid may return undefined in some cases.

https://github.com/react-hook-form/react-hook-form/blob/ad6abef617a8cc25fcda38c93cf64dfb01b0902c/src/logic/createFormControl.ts#L812-L815

Therefore, the following type definition is correct.

```suggestion
export type _UseFormGetFieldState<TFieldValues extends FieldValues> = <
  TFieldName extends FieldPath<TFieldValues>,
>(
  name: TFieldName,
  formState?: FormState<TFieldValues>,
) => {
  invalid: boolean;
  isDirty: boolean | undefined;
  isTouched: boolean | undefined;
  error: FieldError | undefined;
};
```

Reply:
valid point, let me fix it with !!